core/src/main/java/org/mule/runtime/core/routing/IdempotentMessageFilter.java

```diff
@@ -134,6 +128,27 @@ protected boolean accept(Event event, Event.Builder builder) {
     }
   }
 
+  @Override
+  public final Event process(Event event) throws MuleException {
+    if (accept(event)) {
+      return processNext(event);
+    } else {
+      event.getContext().success();
+      return null;
```

Comment:
Whatever we do with other filters and their migration to validators, this processor still needs to be able to drop events.  The mechanism by which this is done in blocking `process` model is by returning null. Echoing the event is not the same behave.